From a4b6a0962fb339d808cfcd2cff007b6ae39e56d8 Mon Sep 17 00:00:00 2001
From: Pontus Fuchs <pontus.fuchs@gmail.com>
Date: Sat, 13 Oct 2012 11:51:39 +0200
Subject: [PATCH 75/82] Use alignment macros

Instead of doing it manually. I also discovered that the skb wasn't properly
extended at the end when doing the alignment.
---
 ar5523.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/drivers/net/wireless/ath/ar5523/ar5523.c
+++ b/drivers/net/wireless/ath/ar5523/ar5523.c
@@ -602,11 +602,12 @@ static void ar5523_data_rx_cb(struct urb
 	skb_put(data->skb, rxlen - sizeof(struct ar5523_rx_desc));
 
 	hdrlen = ieee80211_get_hdrlen_from_skb(data->skb);
-	if (hdrlen & 3) {
+	if (!IS_ALIGNED(hdrlen, 4)) {
 		ar5523_dbg(ar, "eek, alignment workaround activated\n");
-		pad = hdrlen % 4;
+		pad = ALIGN(hdrlen, 4) - hdrlen;
 		memmove(data->skb->data + pad, data->skb->data, hdrlen);
 		skb_pull(data->skb, pad);
+		skb_put(data->skb, pad);
 	}
 
 	rx_status = IEEE80211_SKB_RXCB(data->skb);
